pilot/pkg/model/context.go

```diff
@@ -195,6 +195,10 @@ func (node *Proxy) SetServiceInstances(env *Environment) error {
 		return err
 	}
 
+	if instances == nil || len(instances) == 0 {
```

Comment:
What if this is a workload with no associated service ?

Reply:
it wont work. I dont think its possible currently in istio to run a workload without any service. is it?